solutions/devsprint-michael-douglas-2/DeliveryApp/Screens/Components/LoadingView.swift

```diff
@@ -0,0 +1,95 @@
+//
+//  LoadingView.swift
+//  DeliveryApp
+//
+//  Created by Hyago Henrique on 21/02/22.
+//
+
+import UIKit
+
+final class LoadingView: UIView {
+
+    // MARK: - View Components
+
+    private lazy var loadingContainer: UIView = {
+        let container = UIView(frame: .zero)
+        container.translatesAutoresizingMaskIntoConstraints = false
+        container.isOpaque = true
+        container.isHidden = false
+        container.backgroundColor = .white
+        return container
+    }()
+
+    private lazy var loadingSpinner: UIActivityIndicatorView = {
+        let indicator = UIActivityIndicatorView(frame: .zero)
+        indicator.translatesAutoresizingMaskIntoConstraints = false
+        indicator.style = .large
+        indicator.isHidden = false
+        indicator.hidesWhenStopped = true
+        indicator.color = .black
+        return indicator
+    }()
+
+    private lazy var loadingMessage: UILabel = {
+        let label = UILabel(frame: .zero)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.text = "Carregando..."
+        label.textAlignment = .center
+        label.font = UIFont.systemFont(ofSize: 18, weight: .medium)
+        label.numberOfLines = 0
+        label.textColor = .black
+        return label
+    }()
+
+    // MARK: - Super Methods
+
+    init() {
+        super.init(frame: .zero)
+
+        self.setupViews()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Internal Methods
+    func setLoadingMessage(_ message: String?) {
+        loadingMessage.text = message ?? "Carregando..."
+    }
+    
+    func updateLoading(_ isLoading: Bool) {
+        if isLoading {
+            loadingSpinner.startAnimating()
+        } else {
+            loadingSpinner.stopAnimating()
+        }
+    }
```

Comment:
Como sugestão, mantenha o isHidden aqui:
```suggestion
    func updateLoading(_ isLoading: Bool) {
        isHidden = !isLoading
        
        if isLoading {
            loadingSpinner.startAnimating()
        } else {
            loadingSpinner.stopAnimating()
        }
    }
```

Reply:
Não sabia que o xcode/swift era inteligente o suficiente para eu estando dentro da classe UIView, se eu colocar isHidden = true ele já entender e esconder aquela view... mto legal.